File: week_2/day_08_stack/variations_que/remove_adjacent_duplicates.py
# remove_adjacent_duplicates.py
def removeDuplicates(s):
    stack = []
    for ch in s:
        if stack and stack[-1] == ch:
            stack.pop()
        else:
            stack.append(ch)
    return "".join(stack)

# INPUT
print(removeDuplicates("abbaca"))  # "ca"

File: week_2/day_08_stack/variations_que/removeAdjacentDuplicates.js
// removeAdjacentDuplicates.js
function removeDuplicates(s) {
    const stack = [];
    for (let ch of s) {
        if (stack.length && stack[stack.length - 1] === ch) stack.pop();
        else stack.push(ch);
    }
    return stack.join("");
}

// INPUT
console.log(removeDuplicates("abbaca")); // "ca"
